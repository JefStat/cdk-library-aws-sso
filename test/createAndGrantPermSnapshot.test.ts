import {
  Stack,
  App,
  aws_iam as iam,
  Duration,
} from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import {
  PermissionSet,
  PrincipalTypes,
} from '../src/index';

test('Snapshot', () => {
  const app = new App();
  const stack = new Stack(app, 'TestStack');

  const permissionSet = new PermissionSet(stack, 'PermissionSet', {
    awsManagedPolicies: [
      iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonRDSPerformanceInsightsReadOnly'),
    ],
    description: 'Test Permission Set',
    name: 'TestPermissionSet',
    ssoInstanceArn: 'arn:aws:sso:::instance/ssoins-1234567891234567',
    relayStateType: 'https://us-east-1.console.aws.amazon.com/rds/home',
    sessionDuration: Duration.hours(10),
    customerManagedPolicyReferences: [
      {
        name: 'TestPolicy',
        path: '/',
      },
    ],
    inlinePolicy: new iam.PolicyDocument({
      statements: [
        new iam.PolicyStatement({
          actions: ['rds:*'],
          resources: ['*'],
        }),
      ],
    }),
    permissionsBoundary: {
      managedPolicyArn: iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonRDSPerformanceInsightsReadOnly').managedPolicyArn,
    },
  });

  permissionSet.grant('permissionSetAssignment', {
    principal: {
      principalId: '12350630-0ae9-479a-97c2-0afc2d5b4eac',
      principalType: PrincipalTypes.GROUP,
    },
    targetId: '344567890123456',
  });

  expect(Template.fromStack(stack)).toMatchSnapshot();
});