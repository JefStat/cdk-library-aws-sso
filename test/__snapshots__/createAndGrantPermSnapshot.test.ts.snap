// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "PermissionSetpermissionSetAssignmentassignmentBF53D72E": Object {
      "Properties": Object {
        "InstanceArn": "arn:aws:sso:::instance/ssoins-1234567891234567",
        "PermissionSetArn": Object {
          "Fn::GetAtt": Array [
            "PermissionSetpermissionSetEB39BD79",
            "PermissionSetArn",
          ],
        },
        "PrincipalId": "12350630-0ae9-479a-97c2-0afc2d5b4eac",
        "PrincipalType": "GROUP",
        "TargetId": "344567890123456",
        "TargetType": "AWS_ACCOUNT",
      },
      "Type": "AWS::SSO::Assignment",
    },
    "PermissionSetpermissionSetEB39BD79": Object {
      "Properties": Object {
        "CustomerManagedPolicyReferences": Array [
          Object {
            "Name": "TestPolicy",
            "Path": "/",
          },
        ],
        "Description": "Test Permission Set",
        "InlinePolicy": Object {
          "Statement": Array [
            Object {
              "Action": "rds:*",
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "InstanceArn": "arn:aws:sso:::instance/ssoins-1234567891234567",
        "ManagedPolicies": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/AmazonRDSPerformanceInsightsReadOnly",
              ],
            ],
          },
        ],
        "Name": "TestPermissionSet",
        "PermissionsBoundary": Object {
          "ManagedPolicyArn": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/AmazonRDSPerformanceInsightsReadOnly",
              ],
            ],
          },
        },
        "RelayStateType": "https://us-east-1.console.aws.amazon.com/rds/home",
        "SessionDuration": "PT10H",
      },
      "Type": "AWS::SSO::PermissionSet",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
